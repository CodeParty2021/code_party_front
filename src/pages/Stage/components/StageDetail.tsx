import React from "react";
import { Stage } from "../../../services/StageAPI/StageAPI";

type Prop = {
    stage: Stage | undefined
}

export const StageDetail: React.FC<Prop> = (props: Prop) => {
  if(props.stage === undefined)
    return(
      <>
        情報が見つかりませんでした．
      </>
    )
  return (
    <>
      ID:{props.stage.id},
      NAME:{props.stage.name},
      index: {props.stage.stage_index},
      rule:{props.stage.rule}
    </>
  );
};

export default StageDetail;