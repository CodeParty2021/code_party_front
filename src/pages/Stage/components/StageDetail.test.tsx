import React from "react";
import { shallow } from "enzyme";
import { StageDetail } from "./StageDetail";

describe("<StageDetail />", () => {
  it("Render StageDetail as undefined", () => {
    const wrapper = shallow(<StageDetail stage={undefined} />);
    //expect(wrapper.text()).toEqual("情報が見つかりませんでした．");
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("Render StageDetail", () => {
    const stage = {
      id: 0,
      name: "テストステージ",
      stage_index: 3,
      rule: "テキストテスト",
    };
    const wrapper = shallow(<StageDetail stage={stage} />);
    //expect(wrapper.text()).toEqual("ID:0, NAME:テストステージ, index:3, rule:テキストテスト");
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
