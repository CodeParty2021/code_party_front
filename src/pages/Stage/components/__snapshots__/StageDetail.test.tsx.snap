// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageDetail /> Render StageDetail 1`] = `
Array [
  <React.Fragment>
    ID:
    0
    , NAME:
    テストステージ
    , index:
    3
    , rule:
    テキストテスト
  </React.Fragment>,
]
`;

exports[`<StageDetail /> Render StageDetail as undefined 1`] = `
Array [
  <React.Fragment>
    情報が見つかりませんでした．
  </React.Fragment>,
]
`;
